app/services/FlashSale.scala

```diff
@@ -2,26 +2,46 @@ package services
 
 import configuration.Config
 import model.DigitalEdition
-import model.promoCodes.{GuardianWeekly, _}
+import model.promoCodes.{GuardianWeekly, PaperAndDigital, _}
 import org.joda.time.DateTime
 
-object FlashSale {
+object FlashSale extends FlashSale {
 
-  def inOfferPeriod(promoCodeKey: PromoCodeKey) = {
-    //The offer is valid between 20th August 2018 & 2nd September 2018
-    //The current sale is paper & paper + digital only. Digital only is unaffected.
-    val included: Map[PromoCodeKey, Boolean] = Map(
-      Digital -> false,
-      Paper -> true,
-      PaperAndDigital -> true
+  val startTime = new DateTime(2018, 8, 20, 0, 0)
+  val endTime = new DateTime(2018, 9, 3, 0, 0)
+
+  val flashSaleIntcmp: Map[String, Intcmp] =
+    Map(
+      "GFS80H" -> Intcmp("gdnwb_macqn_other_subs_SupporterLandingPagePrintOnlySubscribeLandingPagePrint+digital_"),
+      "GFS80F" -> Intcmp("gdnwb_macqn_other_subs_guardianarticleSubscribeLandingPagePrintOnly_"),
+      "GFS80K" -> Intcmp("gdnwb_macqn_other_subs_SubPromotionsLandingPagePrintOnlySubsPromotionsLandingPagePrint+digital_"),
+      "GFS80J" -> Intcmp("gdnwb_macqn_other_subs_SupporterLandingPagePrint+digitalSubPromotionsLandingPagePrintOnly_")
     )
+}
+
+trait FlashSale {
 
-    val startTime = new DateTime(2018, 8, 20, 0, 0)
-    val endTime = new DateTime(2018, 9, 3, 0, 0)
+  val startTime: DateTime
+  val endTime: DateTime
+
+  val flashSaleIntcmp: Map[String, Intcmp]
```

Comment:
Minor: it might be easier to understand this if it was a Map[[PromoCode](https://github.com/guardian/membership-common/blob/91523b8ce30603a3b1072e65456a6d237b78b5bf/src/main/scala/com/gu/memsub/promo/Promotion.scala#L18-L20), Intcmp]. To be fair, the example and test code already make this quite simple to understand...